internal/services/keyvault/key_vault_access_policy_data_source.go

```diff
@@ -62,89 +60,53 @@ func dataSourceKeyVaultAccessPolicy() *pluginsdk.Resource {
-		}
-		templateManagementPermissions["secret"] = []string{
-			string(keyvault.SecretPermissionsGet),
-			string(keyvault.SecretPermissionsList),
-			string(keyvault.SecretPermissionsSet),
-			string(keyvault.SecretPermissionsDelete),
-			string(keyvault.SecretPermissionsRecover),
-			string(keyvault.SecretPermissionsBackup),
-			string(keyvault.SecretPermissionsRestore),
-		}
-		templateManagementPermissions["certificate"] = []string{
-			string(keyvault.CertificatePermissionsGet),
-			string(keyvault.CertificatePermissionsList),
-			string(keyvault.CertificatePermissionsUpdate),
-			string(keyvault.CertificatePermissionsCreate),
-			string(keyvault.CertificatePermissionsImport),
-			string(keyvault.CertificatePermissionsDelete),
-			string(keyvault.CertificatePermissionsManagecontacts),
-			string(keyvault.CertificatePermissionsManageissuers),
-			string(keyvault.CertificatePermissionsGetissuers),
-			string(keyvault.CertificatePermissionsListissuers),
-			string(keyvault.CertificatePermissionsSetissuers),
-			string(keyvault.CertificatePermissionsDeleteissuers),
-		}
-	}
+	keyPermissions := flattenKeyPermissions(&[]keyvault.KeyPermissions{
+		keyvault.KeyPermissionsGet,
+		keyvault.KeyPermissionsList,
+		keyvault.KeyPermissionsUpdate,
+		keyvault.KeyPermissionsCreate,
+		keyvault.KeyPermissionsImport,
+		keyvault.KeyPermissionsDelete,
+		keyvault.KeyPermissionsRecover,
+		keyvault.KeyPermissionsBackup,
+		keyvault.KeyPermissionsRestore,
+	})
+
+	secretPermissions := flattenSecretPermissions(&[]keyvault.SecretPermissions{
+		keyvault.SecretPermissionsGet,
+		keyvault.SecretPermissionsList,
+		keyvault.SecretPermissionsSet,
+		keyvault.SecretPermissionsDelete,
+		keyvault.SecretPermissionsRecover,
+		keyvault.SecretPermissionsBackup,
+		keyvault.SecretPermissionsRestore,
+	})
+
+	certificatePermissions := flattenCertificatePermissions(&[]keyvault.CertificatePermissions{
+		keyvault.CertificatePermissionsGet,
+		keyvault.CertificatePermissionsList,
+		keyvault.CertificatePermissionsUpdate,
+		keyvault.CertificatePermissionsCreate,
+		keyvault.CertificatePermissionsImport,
+		keyvault.CertificatePermissionsDelete,
+		keyvault.CertificatePermissionsManagecontacts,
+		keyvault.CertificatePermissionsManageissuers,
+		keyvault.CertificatePermissionsGetissuers,
+		keyvault.CertificatePermissionsListissuers,
+		keyvault.CertificatePermissionsSetissuers,
+		keyvault.CertificatePermissionsDeleteissuers,
+	})
 
 	d.SetId(name)
 
 	if strings.Contains(name, "Key") {
-		d.Set("key_permissions", templateManagementPermissions["key"])
+		d.Set("key_permissions", keyPermissions)
 	}
 	if strings.Contains(name, "Secret") {
-		d.Set("secret_permissions", templateManagementPermissions["secret"])
+		d.Set("secret_permissions", secretPermissions)
 	}
 	if strings.Contains(name, "Certificate") {
-		d.Set("certificate_permissions", templateManagementPermissions["certificate"])
+		d.Set("certificate_permissions", certificatePermissions)
 	}
```

Comment:
we'll need to ensure that a value is always set into the state for these fields, else the list will be `null` rather than an empty slice